### PR TITLE
Rebase DB model class(es) to universal "dashdotdb" class.

### PR DESCRIPTION
**Pull request by mmclanerh**
This is just a WIP/initial pass on the db model class rename, starting with imagemanifestvuln without the new deploymentvalidation bits. If this is solid, will combine +new DV changes on top.

### CODE REVIEW

**Comment by mmclanerh** [comment]
This is looking pretty good now. Imported test data to the ingest endpoint over a few intervals to generate some history, then flipped one of the status from failing to passing and I get back expected summary results. Also verified the logic is sound with regard to minimizing stored data (eg: namespace table postgres=# select * from deploymentvalidation order by token_id;
 id |           name           | token_id | namespace_id | objectkind_id | validation_id 
----+--------------------------+----------+--------------+---------------+---------------
  1 | openshift-acme           |        2 |            2 |             1 |             1
  2 | qontract-api             |        2 |            2 |             1 |             1
  3 | openshift-acme-d6bdb5b48 |        2 |            3 |             2 |             1
  4 | openshift-acme           |        3 |            2 |             1 |             1
  5 | qontract-api             |        3 |            2 |             1 |             1
  6 | openshift-acme-d6bdb5b48 |        3 |            3 |             2 |             1
  7 | openshift-acme           |        4 |            2 |             1 |             1
  8 | qontract-api             |        4 |            2 |             1 |             1
  9 | openshift-acme-d6bdb5b48 |        4 |            3 |             2 |             1
 10 | openshift-acme           |        5 |            2 |             1 |             2
 11 | qontract-api             |        5 |            2 |             1 |             1
 12 | openshift-acme-d6bdb5b48 |        5 |            3 |             2 |             1
(12 rows)

postgres=# select * from namespace;
 id |         name          | cluster_id 
----+-----------------------+------------
  1 | cso                   |          1
  2 | app-interface-stage   |          1
  3 | visual-qontract-stage |          1
(3 rows)

```  

All status(1) totals: curl "localhost:8080/api/v1/metrics"
```
...
# HELP deploymentvalidation_total Validations by validation type
# TYPE deploymentvalidation_total counter
deploymentvalidation_total{cluster="app-sre-prod-01",namespace="app-interface-stage",status="1",validation="deployment_validation_operator_replica_validation"} 2.0
deploymentvalidation_total{cluster="app-sre-prod-01",namespace="visual-qontract-stage",status="1",validation="deployment_validation_operator_replica_validation"} 1.0
...
```  

Demonstrating a validation status transition(0):  curl "localhost:8080/api/v1/deploymentvalidation?cluster=app-sre-prod-01&namespace=app-interface-stage"
```
[
  {
    "context": "openshift-acme",
    "status": 0
  },
  {
    "context": "qontract-api",
    "status": 1
  },
  {
    "context": "openshift-acme-d6bdb5b48",
    "status": 1
  }
]
```  

curl "localhost:8080/api/v1/metrics"
```
# HELP deploymentvalidation_total Validations by validation type
# TYPE deploymentvalidation_total counter
deploymentvalidation_total{cluster="app-sre-prod-01",namespace="app-interface-stage",status="1",validation="deployment_validation_operator_replica_validation"} 1.0
deploymentvalidation_total{cluster="app-sre-prod-01",namespace="visual-qontract-stage",status="1",validation="deployment_validation_operator_replica_validation"} 1.0
deploymentvalidation_total{cluster="app-sre-prod-01",namespace="app-interface-stage",status="0",validation="deployment_validation_operator_replica_validation"} 1.0
```

\o/

File: Makefile [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
-	pip install -r requirements-check.txt
+	pip3 install -r requirements-check.txt
 
 
 develop: clean
 	. venv/bin/activate && \
-	python setup.py develop && \
-	pip install -r requirements-check.txt
+	python3 setup.py develop && \
+	pip3 install -r requirements-check.txt
```
Comment (apahim) [inline Makefile:21]: do you need the `3` sufix even after creating the venv with `python3`?

File: dashdotdb/__init__.py [hunk 2]
```diff
@@ -6,16 +6,16 @@
...
-    DATABASE_HOST = os.environ['DATABASE_HOST']
-    DATABASE_PORT = os.environ['DATABASE_PORT']
-    DATABASE_USERNAME = os.environ['DATABASE_USERNAME']
-    DATABASE_PASSWORD = os.environ['DATABASE_PASSWORD']
-    DATABASE_NAME = os.environ['DATABASE_NAME']
+    DATABASE_HOST = os.getenv('DATABASE_HOST') or 'localhost'
+    DATABASE_PORT = os.getenv('DATABASE_PORT') or '5432'
+    DATABASE_USERNAME = os.getenv('DATABASE_USERNAME') or 'postgres'
+    DATABASE_PASSWORD = os.getenv('DATABASE_PASSWORD') or 'postgres'
+    DATABASE_NAME = os.getenv('DATABASE_NAME') or 'postgres'
```
Comment (apahim) [inline dashdotdb/__init__.py:18]: you have the DASHDOTDB_DATABASE_URL working now and you're exporting it in your `Makefile.devhelpers`. I'd rather have the missing env-var error in production (not having defaults).

File: dashdotdb/__init__.py [hunk 3]
```diff
@@ -27,7 +27,7 @@ class DashDotDb(App):
     def create_app(self):
         # pylint: disable=redefined-outer-name
         app = Flask(self.import_name, **self.server_args)
-
+#       app.config['SQLALCHEMY_ECHO'] = True
```
Comment (apahim) [inline dashdotdb/__init__.py:30]: letft over?

**Review by apahim — CHANGES_REQUESTED** [review]
This looks super good. Missing the ERD update it looks. I haven't validated the accuracy of the reported data, but I expect you to only go ahead when that's checked.

File: Makefile (hunk 1 above)
Comment (mmclanerh) [reply Makefile:21]: hrm. probably not, once venv is loaded. this was mostly for kick-off under a rh7 dev env. i had also looked at updating the makefile a bit and realized it was adding some un-needed scope.. so more or less put things back, but stuck some helpers in a new makefile. i'll revert and push that up.

File: dashdotdb/__init__.py (hunk 2 above)
Comment (mmclanerh) [reply dashdotdb/__init__.py:18]: that is a good point re: erroring if missing..  i'll remove that and push up.

File: dashdotdb/__init__.py (hunk 3 above)
Comment (mmclanerh) [reply dashdotdb/__init__.py:30]: yep. was thinking: it'd be helpful to add a debug toggle here, but again: more (un-necessary) scope. will clean up.

File: dashdotdb/__init__.py (hunk 2 above)
Comment (mmclanerh) [reply dashdotdb/__init__.py]: reverted back to os.environ[]

File: dashdotdb/__init__.py (hunk 3 above)
Comment (mmclanerh) [reply dashdotdb/__init__.py]: removed. did note in readme how to enable.. because knowing me, i'll probably forget..

File: Makefile (hunk 1 above)
Comment (mmclanerh) [reply Makefile]: reverted /3//

**Comment by mmclanerh** [comment]
> This looks super good. Missing the ERD update it looks. I haven't validated the accuracy of the reported data, but I expect you to only go ahead when that's checked.

Updated .dia/.png. added some of the relationship naming, to assist the next contributor.  verified a few different cases of adding/ re-adding/removing/etc data blobs as well as verified the relationships for the vuln data are still adding/returning what they should..  and it all returns as expected. :+1:

**Review by apahim — APPROVED** [review]
LGTM, thank you for all the work here.

Those blue annotations on the ERD, are they leftovers? If not, I'd suggest having them to all the tables OR not having them at all... it makes the whole thing a bit harder to maintain, but I'm ok with keeping them if we complete all the tables.

Upgrade path seems to work well. Please let it chill for a bit in stage, I want to run the CSO client against this new version on stage before we promote to prod. I will do it right after you merge.

Do you have the client bits already? Please use stage to validate that as weel. From my experience, code will change a couple of times when we start testing the client :)

File: Makefile [hunk 4]
```diff
@@ -16,7 +18,7 @@ install-requirements: install
 develop: clean
 	. venv/bin/activate && \
 	python setup.py develop && \
-	pip install -r requirements-check.txt
+	pip install -r requirements-check.txt
```
Comment (rporres) [inline Makefile:21]: nit: trailing whitespace

**Comment by rporres** [comment]
It looks good to me given the extent of my dashdotdb knowledge 😅

**Comment by mmclanerh** [comment]
going to get this into stage so we can check out the whole picture. i'll submit a separate mr to update the diagram based on the feedback..  i was looking at converting that over to uml links, so we could define the references properly, then realized we could probably take the code and auto-generate the diagram, or most of it..  so again, will work on that in it's own mr.

**Pull request merged**